src/api-extension/cartActions/addDirectDiscountWithDiscountSummary.ts

```diff
@@ -6,27 +6,60 @@ export default function addDirectDiscountWithDiscountSummary(
   cart: Cart,
   validateCouponsResult: ValidateCouponsResult,
 ): CartActionSetDirectDiscounts[] {
-  const { totalDiscountAmount, applicableCoupons } = validateCouponsResult;
+  const { applicableCoupons } = validateCouponsResult;
   if (applicableCoupons.length === 0) return [];
   const { currencyCode } = cart.totalPrice;
 
-  return [
-    {
-      action: 'setDirectDiscounts',
-      discounts: [
-        {
-          target: { type: 'lineItems', predicate: 'true' },
+  const discounts = [];
+
+  applicableCoupons.forEach((coupon) => {
+    if (coupon.result.discount.effect === 'APPLY_TO_ORDER') {
```

Comment:
if the discount effect is `APPLY_TO_ORDER`, then you add the discount twice? (as there is no `return` or `else` statement)